core/javax.cache/src/main/java/org/wso2/carbon/caching/impl/CacheImpl.java

```diff
@@ -124,6 +134,9 @@ public CacheImpl(String cacheName, CacheManager cacheManager) {
                     Util.getDistributedMapNameOfCache(CachingConstants.TIMESTAMP_CACHE_PREFIX +
                             cacheName, ownerTenantDomain, cacheManager.getName()), new TimestampMapEntryListenerImpl());
         }
+
+        cacheEntryListeners.add(clusterCacheInvalidationReqSender);
```

Comment:
Why do you need this? When remove is called on the CacheImpl object you can simply do the cache invalidation messaging.